workers/unity/Packages/com.improbable.gdk.tools/DownloadCoreSdk.cs

```diff
@@ -133,7 +141,92 @@ private static DownloadResult Download()
                 EditorApplication.UnlockReloadAssemblies();
             }
 
-            return exitCode == 0 ? DownloadResult.Success : DownloadResult.Error;
+            if (exitCode != 0)
+            {
+                return DownloadResult.Error;
+            }
+
+            AssetDatabase.Refresh();
+            SetPluginsCompatibility();
+            return DownloadResult.Success;
+        }
+
+        /// <summary>
+        ///     Sets plugin platform compatibility based on directory structure
+        /// </summary>
+        private static void SetPluginsCompatibility()
+        {
+            foreach (var pluginDirectoryCompatibility in PluginsCompatibilityList)
+            {
+                var pluginPaths = AssetDatabase.FindAssets("", new[] { pluginDirectoryCompatibility.Path });
+                foreach (var pluginPath in pluginPaths)
```

Comment:
How about `foreach (var pluginPath in pluginPaths.OfType<PluginImporter>())`?
Then you can drop the first chunk of the loop.

Reply:
(y)

Reply:
Just realised that this won't work, because we only have asset GUID, not asset itself from AssetDatabase - we can't distinguish between plugin and non-plugin until we get actual instance